src/providers/mod.rs

```diff
@@ -132,6 +142,16 @@ pub trait Provide {
         Err(ResponseStatus::PsaErrorNotSupported)
     }
 
+    /// Lists all keys belonging to the application.
+    fn list_keys(
```

Comment:
I think it would be nicer to only have the `list_keys` method and:
* in all the crypto providers make it the same as what `find_keys` currently does
* in the code provider call `list_keys` on all provider

Although we need to be sure somewhere that `ListKeys` can not be called directly on the crypto providers. Probably somewhere higher on the stack.